### ISSUE TITLE
Swift Assertions in Xcode 9.1

### ISSUE THREAD

**Issue by blessingLopes** [Quick/Nimble#478]
### What did you do?

I used  `expect { () -> Void in fatalError() }.to(throwAssertion())` in a test, which
throws a fatal error, completely stops the tests suite execution, and we get a
>  expected to raise any exception, got no exception 

test failed message

### What did you expect to happen?

 `expect { () -> Void in fatalError() }.to(throwAssertion())`  to succeed.

### What actually happened instead?

`expect { () -> Void in fatalError() }.to(throwAssertion())`  fails.

### Environment

List the software versions you're using:

 - Nimble: *7.0*
 - Xcode Version: *9.1 (9B55)* 
 - Swift Version: *4.0*

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Cocoapods: *1.3.1*

Maybe I'm missing something or doing something wrong, my deep apologies if that's the case.
Thanks for any feedback!

**Comment by szweier** [comment]
Seeing the same issue seems to be Xcode 9.1 specific.

**Comment by VojtaStavik** [comment]
This is happening only when the test suite is ran in the host application. For standalone tests everything works OK.

**Comment by bryantjustin** [comment]
You can continue the tests by clicking on "Continue program execution" but this gets tedious if you have a few of these type of tests.

**Comment by msewell** [comment]
As a sort-of workaround for getting stuck on debugger breakpoints for fatal errors when running tests, you can run your tests without debugging support. Disable 'Debug Executable' in your scheme by going to _Edit Scheme > Test > Info_ and unchecking 'Debug Executable'.

**Comment by blessingLopes** [comment]
Many thanks for the feedback and useful suggestions!

**Comment by MortyMerr** [comment]
Maybe, you should mention this in documentation? Solution was not obvious for me.

**Comment by msewell** [comment]
@MortyMerr: Disabling debugging during test runs isn't really a solution (it hardly even is a workaround). This issue still needs a "real" fix.

**Comment by rafaelnobrepd** [comment]
A side effect of disable debugging during tests (which is quite inconvenient as is), is that it seems to affect coverage reports (always reports 0% when debug executable is disabled).
Unfortunately its a real pain, to the point that I'm considering removing the assertion expectations

**Comment by mvandervelden** [comment]
Haven't had this issue before, until I started using Xcode 9.3 / Swift 4.1, where it fails consistently. Can confirm that disabling debugging during tests works, but as mentioned before, it's not a suitable workaround.

**Comment by RetepV** [comment]
Woops. And I am finding this issue as well, after having upgraded to Xcode 9.3.

```
            it("Assertion from inch unit test") {
               expect {
                    try HeightConverter.stringForProfileHeightUnit(HeightUnit.inch)
                }.to(throwAssertion())
            }
```

Code to test executes the following assert:

```
            assert(false, "Profile only defines cm or ft. Anything else is a bug.")
```

In Xcode 9.2, the test pass fine. In Xcode 9.3, the line actually asserts and the test run fails because it detects an assertion (which it's actually supposed to detect).

**Comment by drekka** [comment]
Hitting this with 9.3 as well.

**Comment by danielaRiesgo** [comment]
Hello. Just wanted to know if there is an update about this?

I'm having the same problem. Worked in XCode 9.2 but not in XCode 9.3.

**Comment by drekka** [comment]
Problem is persisting in Xcode 9.4. I suspect this will not be fixed and Nimble will have to find an alternative solution. For myself, I've gone with the approach of using a local override of `fatalError(...)` and then adding code to set a mock equivalent during testing. Unfortunately I cannot publish the code here as I did it for a client and they would have to approve it (unlikely!), however I can say it was based on https://medium.com/@marcosantadev/how-to-test-fatalerror-in-swift-e1be9ff11a29

**Comment by ikesyo** [comment]
Looks like we can adopt https://github.com/mattgallagher/CwlPreconditionTesting/commit/ce96cb1d81644b92c66fd2b622260905526a7c5f to address the issue.

Related to https://github.com/apple/swift/pull/11329.

**Comment by ikesyo** [comment]
Fixed by #545.
